src/serverless/structure/serverless_parser.go

```diff
@@ -0,0 +1,326 @@
+package structure
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/awslabs/goformation/v4/cloudformation"
+	"github.com/bridgecrewio/yor/src/common"
+	"github.com/bridgecrewio/yor/src/common/logger"
+	"github.com/bridgecrewio/yor/src/common/structure"
+	"github.com/bridgecrewio/yor/src/common/tagging/tags"
+	"github.com/bridgecrewio/yor/src/common/types"
+	"github.com/bridgecrewio/yor/src/common/utils"
+	yamlUtils "github.com/bridgecrewio/yor/src/common/yaml"
+	"gopkg.in/yaml.v2"
+)
+
+const FunctionTagsAttributeName = "tags"
+const StackTagsAttributeName = "stackTags"
+
+var TemplateSections = []string{"service", "provider", "tags", "stackTags", "resources", "functions", "region", "Resources"}
+
+type ServerlessTemplate struct {
+	Service  string `yaml:"service"`
+	Provider struct {
+		Name         string            `yaml:"name"`
+		Runtime      string            `yaml:"runtime"`
+		Region       string            `yaml:"region"`
+		ProviderTags map[string]string `yaml:"tags"`
+		CFNTags      map[string]string `yaml:"stackTags"`
+	} `yaml:"provider"`
+	Functions interface{} `yaml:"functions"`
+	Resources struct {
+		Resources *cloudformation.Template `yaml:"Resources"`
+	} `yaml:"resources"`
+}
+
+type ServerlessParser struct {
+	YamlParser types.YamlParser
+	Template   *ServerlessTemplate
+}
+
+func (p *ServerlessParser) Init(rootDir string, _ map[string]string) {
+	p.YamlParser.RootDir = rootDir
+	p.YamlParser.FileToResourcesLines = make(map[string]structure.Lines)
+	p.Template = &ServerlessTemplate{}
+}
+
+func (p *ServerlessParser) GetSkippedDirs() []string {
+	return []string{}
+}
+
+func (p *ServerlessParser) GetSupportedFileExtensions() []string {
+	return []string{common.YamlFileType.Extension, common.YmlFileType.Extension}
+}
+
+func (p *ServerlessParser) ParseFile(filePath string) ([]structure.IBlock, error) {
+	parsedBlocks := make([]structure.IBlock, 0)
+	fileFormat := utils.GetFileFormat(filePath)
+	fileName := filepath.Base(filePath)
+	if !(fileName == fmt.Sprintf("serverless.%s", fileFormat)) {
+		return nil, nil
+	}
+	// #nosec G304 - file is from user
+	template, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		logger.Warning(fmt.Sprintf("There was an error processing the serverless template: %s", err))
+	}
+	err = yaml.Unmarshal(template, p.Template)
+	if err != nil {
+		logger.Error(fmt.Sprintf("Unmarshal: %s", err), "SILENT")
+	}
+	if p.Template.Functions == nil && p.Template.Resources.Resources == nil {
+		return parsedBlocks, nil
+	}
+
+	if err != nil || template == nil {
+		logger.Error(fmt.Sprintf("There was an error processing the serverless template: %s", err), "SILENT")
+
+	}
+	// cfnStackTagsResource := p.template.Provider.CFNTags
```

Comment:
@tronxd remove this commented out line please

Reply:
It's for the next PR :)